Objects/weakrefobject.c

```diff
@@ -1018,3 +1018,19 @@ PyObject_ClearWeakRefs(PyObject *object)
         PyErr_Restore(err_type, err_value, err_tb);
     }
 }
+
+/* This function is called by _PyStaticType_Dealloc() to clear weak references.
+ *
+ * This is called at the end of runtime finalization, so we can just
+ * wipe out the type's weaklist.  We don't bother with callbacks
+ * or anything else.
+ */
+void
+_PyStaticType_ClearWeakRefs(PyTypeObject *type)
+{
+    static_builtin_state *state = _PyStaticType_GetState(type);
+    PyObject **list = _PyStaticType_GET_WEAKREFS_LISTPTR(state);
+    while (*list != NULL) {
```

Comment:
This should be `if` rather than `while`, no?

Reply:
I agree it is confusing.  However, it is how `clear_weakref()` works.  `*list` is a pointer to the head of the linked list and `list` is a pointer to that pointer.  `clear_weakref()` actually modifies `list`, changing where `*list` points.  Thus each time through the loop we are actually popping off the next node and clearing it.

A short comment here wouldn't hurt, even if `clear_weakref()` is fundamentally confusing everywhere.